### ISSUE TITLE
`cssStyleBindings` Proposal

### ISSUE THREAD

**Issue by mattsacks** [salsify/ember-css-modules#26]
Hey there!

We've been using this project at @UseCanvas and thought we could make it a bit easier to add root component styles based on property truthiness, similar to how `classNameBindings` works.

What do you think about adding a formal property for this? Here's an example of how `cssStyleBindings` would work:

```javascript
export default Ember.Component.extend({
  cssStyleBindings: [
     'component', 'isShowing:is-showing', 'isAvailable:available:unavailable'
  ]
});
```
```html
<div class="ember-view _component_oksdf _is-showing_lkjsdfl _unavailable_lkjsdf">...</div>
```

/cc @max

**Comment by dfreeman** [comment]
This definitely feels like a useful addition. A couple questions:
 - In your example above, would the `component` class always be applied, or only if a corresponding `component` property has a truthy value? Based on the regular `classNameBindings` behavior, I'd expect it to be the latter, but it also seems like there are cases where you always want to apply certain classes. Do you think a second configuration property (akin to `classNames` vs `classNameBindings`) makes sense?
 - The name `cssStyleBindings` seems at first glance like it would do something more like [ember-style-bindings](https://github.com/andrewtimberlake/ember-style-bindings). In teaching these concepts and thinking about them, I've been referring to a component's namespaced class names as "local classes". What do you think about something like `localClassNameBindings` (and possibly `localClassNames`, depending on the previous question) – too verbose?

Once we get those two points hammered out, would your or @max want to take a stab at a PR?

**Comment by max** [comment]
👍 on @mattsacks and I workinng on the PR.

**Comment by max** [comment]
@dfreeman I really like `localClassNames` and `localClassNameBindings`. It makes it very easy to teach. We'll experiment with both of those.

**Comment by alvinvogelzang** [comment]
@max how are things going with the PR? `localClassNames` and `localClassNameBindings` is much more meaningful indeed :-)

**Comment by alvinvogelzang** [comment]
bump?

**Comment by ynnoj** [comment]
Dying to see this land 🚀

**Comment by mattsacks** [comment]
Hey all! Glad to see that there's excitement for this 😀 

I cooked up [a commit that adds `localClassNameBindings` to Ember.Component](#29), but it still needs some work. A quick glance: 

* Support for `localClassNames`, a static list of classes that are scoped for the component (via `this.get('styles.{localClassName}')`.

* Review! I'm new to writing Ember mixins and I'm not sure I'm doing things "the right way". For instance, I'm overwriting `classNames` on `init()` and `didUpdateAttrs()`, which is probably a bad idea. What would be the best way for this feature to work? Should it clone the initial `classNames` and reference that?

* Tests! I'm completely unfamiliar to testing in Ember and could really use some pointers for how this feature could be tested

Thanks all for the encouragement to keep working on this!

**Comment by dfreeman** [comment]
Just released v0.4.3 with support for `localClassNames` and `localClassNameBindings`, so I'm going to go ahead and close this out. Thanks for your help defining this feature!
